### ISSUE TITLE
Build error on Arch Linux

### ISSUE THREAD

**Issue by basus** [mirage/ocaml-cohttp#633]
I'm getting the following error when trying to build `cohttp-async` on Arch Linux with OCaml 4.06.1:

```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of cohttp-async failed at "jbuilder build -p cohttp-async -j 1".

#=== ERROR while installing cohttp-async.1.0.2 ================================#
# opam-version 1.2.2
# os           linux
# command      jbuilder build -p cohttp-async -j 1
# path         /home/basus/.opam/4.06.1/build/cohttp-async.1.0.2
# compiler     4.06.1
# exit-code    1
# env-file     /home/basus/.opam/4.06.1/build/cohttp-async.1.0.2/cohttp-async-691-1872ec.env
# stdout-file  /home/basus/.opam/4.06.1/build/cohttp-async.1.0.2/cohttp-async-691-1872ec.out
# stderr-file  /home/basus/.opam/4.06.1/build/cohttp-async.1.0.2/cohttp-async-691-1872ec.err
### stderr ###
# [...]
# collect2: error: ld returned 1 exit status
# File "caml_startup", line 1:
# Error: Error during linking
#     ocamlopt cohttp-async/bin/cohttp_server_async.exe (exit 2)
# (cd _build/default && /home/basus/.opam/4.06.1/bin/ocamlopt.opt -w -40 -g -o cohttp-async/bin/cohttp_server_async.exe -I /home/basus/.opam/4.06.1/lib/astring -I /home/basus/.opam/4.06.1/lib/async -I /home/basus/.opam/4.06.1/lib/async_extra -I /home/basus/.opam/4.06.1/lib/async_kernel -I /home/basus/.opam/4.06.1/lib/async_rpc_kernel -I /home/basus/.opam/4.06.1/lib/async_unix -I /home/basus/.opam/4.06.1/lib/base -I /home/basus/.opam/4.06.1/lib/base/caml -I /home/basus/.opam/4.06.1/lib/base/md5 -I /home/basus/.opam/4.06.1/lib/base/shadow_stdlib -I /home/basus/.opam/4.06.1/lib/base64 -I /home/basus/.opam/4.06.1/lib/bin_prot -I /home/basus/.opam/4.06.1/lib/bin_prot/shape -I /home/basus/.opam/4.06.1/lib/bytes -I /home/basus/.opam/4.06.1/lib/cohttp -I /home/basus/.opam/4.06.1/lib/conduit -I /home/basus/.opam/4.06.1/lib/conduit-async -I /home/basus/.opam/4.06.1/lib/core -I /home/basus/.opam/4.06.1/lib/core_kernel -I /home/basus/.opam/4.06.1/lib/core_kernel/base_for_tests -I /home/basus/.opam/4.06.1/lib/fieldslib -I /home/basus/.opam/4.06.1/lib/fmt -I /home/basus/.opam/4.06.1/lib/ipaddr -I /home/basus/.opam/4.06.1/lib/ipaddr/unix -I /home/basus/.opam/4.06.1/lib/jane-street-headers -I /home/basus/.opam/4.06.1/lib/logs -I /home/basus/.opam/4.06.1/lib/magic-mime -I /home/basus/.opam/4.06.1/lib/ocaml/threads -I /home/basus/.opam/4.06.1/lib/parsexp -I /home/basus/.opam/4.06.1/lib/ppx_assert/runtime-lib -I /home/basus/.opam/4.06.1/lib/ppx_bench/runtime-lib -I /home/basus/.opam/4.06.1/lib/ppx_compare/runtime-lib -I /home/basus/.opam/4.06.1/lib/ppx_expect/collector -I /home/basus/.opam/4.06.1/lib/ppx_expect/common -I /home/basus/.opam/4.06.1/lib/ppx_expect/config -I /home/basus/.opam/4.06.1/lib/ppx_hash/runtime-lib -I /home/basus/.opam/4.06.1/lib/ppx_inline_test/config -I /home/basus/.opam/4.06.1/lib/ppx_inline_test/runtime-lib -I /home/basus/.opam/4.06.1/lib/ppx_sexp_conv/runtime-lib -I /home/basus/.opam/4.06.1/lib/protocol_version_header -I /home/basus/.opam/4.06.1/lib/re -I /home/basus/.opam/4.06.1/lib/re/posix -I /home/basus/.opam/4.06.1/lib/result -I /home/basus/.opam/4.06.1/lib/seq -I /home/basus/.opam/4.06.1/lib/sexplib -I /home/basus/.opam/4.06.1/lib/sexplib/unix -I /home/basus/.opam/4.06.1/lib/sexplib0 -I /home/basus/.opam/4.06.1/lib/spawn -I /home/basus/.opam/4.06.1/lib/splittable_random -I /home/basus/.opam/4.06.1/lib/stdio -I /home/basus/.opam/4.06.1/lib/stringext -I /home/basus/.opam/4.06.1/lib/typerep -I /home/basus/.opam/4.06.1/lib/uchar -I /home/basus/.opam/4.06.1/lib/uri -I /home/basus/.opam/4.06.1/lib/uri/services -I /home/basus/.opam/4.06.1/lib/variantslib -I cohttp-async/src -I cohttp_server /home/basus/.opam/4.06.1/lib/result/result.cmxa /home/basus/.opam/4.06.1/lib/logs/logs.cmxa /home/basus/.opam/4.06.1/lib/fmt/fmt.cmxa /home/basus/.opam/4.06.1/lib/logs/logs_fmt.cmxa /home/basus/.opam/4.06.1/lib/base/caml/caml.cmxa /home/basus/.opam/4.06.1/lib/base/shadow_stdlib/shadow_stdlib.cmxa /home/basus/.opam/4.06.1/lib/sexplib0/sexplib0.cmxa /home/basus/.opam/4.06.1/lib/base/base.cmxa /home/basus/.opam/4.06.1/lib/base/md5/md5_lib.cmxa /home/basus/.opam/4.06.1/lib/ocaml/unix.cmxa /home/basus/.opam/4.06.1/lib/ocaml/bigarray.cmxa /home/basus/.opam/4.06.1/lib/fieldslib/fieldslib.cmxa /home/basus/.opam/4.06.1/lib/ppx_compare/runtime-lib/ppx_compare_lib.cmxa /home/basus/.opam/4.06.1/lib/ppx_sexp_conv/runtime-lib/ppx_sexp_conv_lib.cmxa /home/basus/.opam/4.06.1/lib/variantslib/variantslib.cmxa /home/basus/.opam/4.06.1/lib/bin_prot/shape/bin_shape_lib.cmxa /home/basus/.opam/4.06.1/lib/bin_prot/bin_prot.cmxa /home/basus/.opam/4.06.1/lib/ppx_hash/runtime-lib/ppx_hash_lib.cmxa /home/basus/.opam/4.06.1/lib/ppx_inline_test/config/inline_test_config.cmxa /home/basus/.opam/4.06.1/lib/ppx_inline_test/runtime-lib/ppx_inline_test_lib.cmxa /home/basus/.opam/4.06.1/lib/core_kernel/base_for_tests/base_for_tests.cmxa /home/basus/.opam/4.06.1/lib/jane-street-headers/jane_street_headers.cmxa /home/basus/.opam/4.06.1/lib/ppx_assert/runtime-lib/ppx_assert_lib.cmxa /home/basus/.opam/4.06.1/lib/ppx_bench/runtime-lib/ppx_bench_lib.cmxa /home/basus/.opam/4.06.1/lib/ppx_expect/common/expect_test_common.cmxa /home/basus/.opam/4.06.1/lib/ppx_expect/config/expect_test_config.cmxa /home/basus/.opam/4.06.1/lib/ppx_expect/collector/expect_test_collector.cmxa /home/basus/.opam/4.06.1/lib/parsexp/parsexp.cmxa /home/basus/.opam/4.06.1/lib/sexplib/sexplib.cmxa /home/basus/.opam/4.06.1/lib/typerep/typerep_lib.cmxa /home/basus/.opam/4.06.1/lib/splittable_random/splittable_random.cmxa /home/basus/.opam/4.06.1/lib/stdio/stdio.cmxa /home/basus/.opam/4.06.1/lib/core_kernel/core_kernel.cmxa /home/basus/.opam/4.06.1/lib/async_kernel/async_kernel.cmxa /home/basus/.opam/4.06.1/lib/protocol_version_header/protocol_version_header.cmxa /home/basus/.opam/4.06.1/lib/async_rpc_kernel/async_rpc_kernel.cmxa /home/basus/.opam/4.06.1/lib/sexplib/unix/sexplib_unix.cmxa /home/basus/.opam/4.06.1/lib/ocaml/threads/threads.cmxa /home/basus/.opam/4.06.1/lib/spawn/spawn.cmxa /home/basus/.opam/4.06.1/lib/core/core.cmxa /home/basus/.opam/4.06.1/lib/async_unix/async_unix.cmxa /home/basus/.opam/4.06.1/lib/async_extra/async_extra.cmxa /home/basus/.opam/4.06.1/lib/async/async.cmxa /home/basus/.opam/4.06.1/lib/seq/seq.cmxa /home/basus/.opam/4.06.1/lib/re/re.cmxa /home/basus/.opam/4.06.1/lib/re/posix/re_posix.cmxa /home/basus/.opam/4.06.1/lib/stringext/stringext.cmxa /home/basus/.opam/4.06.1/lib/uri/uri.cmxa /home/basus/.opam/4.06.1/lib/uri/services/uri_services.cmxa /home/basus/.opam/4.06.1/lib/ipaddr/ipaddr.cmxa /home/basus/.opam/4.06.1/lib/ipaddr/unix/ipaddr_unix.cmxa /home/basus/.opam/4.06.1/lib/astring/astring.cmxa /home/basus/.opam/4.06.1/lib/conduit/conduit.cmxa /home/basus/.opam/4.06.1/lib/conduit-async/conduit_async.cmxa /home/basus/.opam/4.06.1/lib/magic-mime/magic_mime_library.cmxa /home/basus/.opam/4.06.1/lib/base64/base64.cmxa /home/basus/.opam/4.06.1/lib/cohttp/cohttp.cmxa cohttp-async/src/cohttp_async.cmxa cohttp_server/cohttp_server.cmxa /home/basus/.opam/4.06.1/lib/fmt/fmt_tty.cmxa cohttp-async/bin/.cohttp_curl_async.eobjs/cohttp_server_async.cmx)
# /usr/sbin/ld: /home/basus/.opam/4.06.1/lib/core/libcore_stubs.a(unix_stubs.o): in function `unix_mknod_stub':
# /home/basus/.opam/4.06.1/build/core.v0.11.2/_build/default/src/unix_stubs.c:281: undefined reference to `makedev'
# collect2: error: ld returned 1 exit status
# File "caml_startup", line 1:
# Error: Error during linking
```

Please let me know if there's any more information I can provide.

**Comment by mseri** [comment]
It will peobably require a new release of core. Does it compile if you pin it to the dev version (opam pin add core —dev-repo)

**Comment by basus** [comment]
I'm trying that now. OPAM (or rather, `git`, I think) complains that it can't find a remote helper for `git+https` so I'm trying to figure out how to solve that first.

**Comment by rjbou** [comment]
It is a know _issue_, the syntax `git+https` was introduced on the new opam2.0 format and you seem using opam1.2. 
However, you can pin the development version with 
```
opam pin add core https://github.com/janestreet/core -k git
```
or [upgrade to opam2.0](https://opam.ocaml.org/blog/opam-2-0-0-rc4/) ;)

**Comment by basus** [comment]
Ah thanks! I can now manage to get OPAM to use the correct repo, but building Core is failing with:

```


#=== ERROR while installing core.v0.11.2 ======================================#
# opam-version 1.2.2
# os           linux
# command      jbuilder build -p core -j 1
# path         /home/basus/.opam/4.06.1/build/core.v0.11.2
# compiler     4.06.1
# exit-code    1
# env-file     /home/basus/.opam/4.06.1/build/core.v0.11.2/core-14434-2c871f.env
# stdout-file  /home/basus/.opam/4.06.1/build/core.v0.11.2/core-14434-2c871f.out
# stderr-file  /home/basus/.opam/4.06.1/build/core.v0.11.2/core-14434-2c871f.err
### stderr ###
# File "src/jbuild", line 4, characters 16-107:
# Error: Public library "core_kernel.config_h" not found
```

I've [filed an issue](https://github.com/janestreet/core/issues/118) with Core.

**Comment by basus** [comment]
This seems to be resolved with OPAM 2.0 and Core v0.11.3, so I'm closing this issue.
